src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/bolt/SwitchManagerHub.java

```diff
@@ -120,20 +121,30 @@ public SwitchManagerHub(HubBolt.Config hubConfig, PersistenceManager persistence
     public void init() {
         super.init();
 
-        validateService = new SwitchValidateServiceImpl(this, persistenceManager,
-                new ValidationServiceImpl(persistenceManager),
-                new RuleManagerImpl(ruleManagerConfig));
-        syncService = new SwitchSyncServiceImpl(this, persistenceManager, flowResourcesConfig);
-        switchRuleService = new SwitchRuleServiceImpl(this, persistenceManager.getRepositoryFactory());
-
         LagPortOperationConfig config = new LagPortOperationConfig(
                 persistenceManager.getRepositoryFactory(), persistenceManager.getTransactionManager(),
                 topologyConfig.getBfdPortOffset(), topologyConfig.getBfdPortMaxNumber(),
                 topologyConfig.getLagPortOffset(), topologyConfig.getLagPortMaxNumber(),
                 topologyConfig.getLagPortPoolChunksCount(), topologyConfig.getLagPortPoolCacheSize());
         log.info("LAG logical ports service config: {}", config);
-        createLagPortService = new CreateLagPortServiceImpl(this, config);
-        deleteLagPortService = new DeleteLagPortServiceImpl(this, config);
+
+        timeoutDispatchMap = new HashMap<>();
+
+        serviceRegistry = new HashMap<>();
+        validateService = registerService(serviceRegistry, "switch-validate", this,
+                carrier -> new SwitchValidateService(
+                        carrier, persistenceManager,
+                        new ValidationServiceImpl(persistenceManager), new RuleManagerImpl(ruleManagerConfig)));
+        syncService = registerService(
+                serviceRegistry, "switch-sync", this,
```

Comment:
maybe constants or enum?

Reply:
No. At least now. This is one time used value. The only requirement for them is too be unique on this specific layer (class). Because it is one time used value, I don't want to make a "constant" for them. Mostly same reason to not make an enum type here.

Reply:
at least leave a todo here, because it looks weird